allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,245 @@
+    "TRAINING_3_images_B_10uL_reward",
+    "TRAINING_3_images_G_10uL_reward",
+    "TRAINING_4_images_A_handoff_lapsed",
+    "TRAINING_4_images_A_handoff_ready",
+    "TRAINING_4_images_A_training",
+    "TRAINING_4_images_B_training",
+    "TRAINING_4_images_G_training",
+    "TRAINING_5_images_A_epilogue",
+    "TRAINING_5_images_A_handoff_lapsed",
+    "TRAINING_5_images_A_handoff_ready",
+    "TRAINING_5_images_B_epilogue",
+    "TRAINING_5_images_B_handoff_lapsed",
+    "TRAINING_5_images_B_handoff_ready",
+    "TRAINING_5_images_G_epilogue",
+    "TRAINING_5_images_G_handoff_lapsed",
+    "TRAINING_5_images_G_handoff_ready",
+]
+
+PROJECT_CODES = [
+    "VisualBehavior",
+    "VisualBehaviorTask1B",
+    "VisualBehaviorMultiscope",
+    "VisualBehaviorMultiscope4areasx2d",
+]
+
+NUM_STRUCTURES_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 2,
+    "VisualBehaviorMultiscope4areasx2d": 4,
+}
+
+NUM_DEPTHS_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 4,
+    "VisualBehaviorMultiscope4areasx2d": 2,
+}
+
+OPHYS_BEHAV_TYPE_DICT = {
+    "active_behavior": ["0", "1", "3", "4", "6"],
+    "passive_viewing": ["2", "5", "7"],
+}
+
+
+############################
+#
+#     from session type
+#
+############################
+
+
+def parse_behavior_context(metadata_row: pd.Series) -> Union[str, None]:
+    """Return if session is passive or active.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    behavior_type : str
+        Return string describing active or passive session.
+    """
+    passive_ophys_session_nums = ["2", "5", "7"]
+
+    session_type = metadata_row["session_type"]
+
+    if session_type not in SESSION_TYPES:
+        return None
+    is_passive = (
+        any(
+            passive_num in session_type
+            for passive_num in passive_ophys_session_nums
+        )
+        and "OPHYS" in session_type
+    )
+    if is_passive:
+        behavior_type = "passive_viewing"
+    else:
+        behavior_type = "active_behavior"
+    return behavior_type
+
+
+def parse_stimulus_set(metadata_row: pd.Series) -> Union[str, None]:
+    """Return the name of the image set or gratings.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    stim_set : str
+        Name of stimulus set shown for session.
+    """
+    session_type = metadata_row["session_type"]
+    if session_type not in SESSION_TYPES:
+        return None
+
+    split_str = session_type.split("_")
+    if split_str[2] == "images":
+        stim_set = split_str[2] + "_" + split_str[3]
+    elif split_str[2] == "gratings":
+        stim_set = "gratings"
+    return stim_set
+
+
+def parse_stimulus_set_novelty(metadata_row: pd.Series) -> Union[str, None]:
```

Comment:
This is already calculated here https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/brain_observatory/vbn_2022/metadata_writer/dataframe_manipulations.py#L241

Scott wrote this for VBN. Please refactor this code so that it is usable here as well.

Reply:
Imported that function you linked and am now using it directly instead of the additional parser.